### ISSUE TITLE
[a11y]: SideNavMenuItem is not  keyboard operable, unable to focus using tab

### ISSUE THREAD

**Issue by lisawu-1** [carbon-design-system/carbon#16083]
### Package

@carbon/react

### Operating System

MacOS

### Package version

1.50.0

### React version

17.0.2

### Automated testing tool and ruleset

https://www.ibm.com/able/requirements/requirements/?version=v7_2#1_4_13

### Description

Expected: Able use tab to focus on each <SideNavMenuItem> under <SideNavMenu>

### Reproduction/example

<SideNavMenuItem> not tabbable

### Steps to reproduce

Use tab to go through the webpage. Unable to focus on SideNavMenuItem

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by Gururajj77** [comment]
Hi! could you please reproduce the persisting issue on stackblitz, so that we can debug further? Here is a link of the stackblitz: https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-18-vite/?file=src%2Findex.scss,src%2FApp.jsx&preset=node=


If you are using any other react and carbon version, please use the below sandboxes matching your version to reproduce the issue:

https://github.com/carbon-design-system/sandboxes

**Comment by nkn2022** [comment]
Switching from `SideNavMenuItem` to `SideNavLink` resolved this issue for me. May be used as a temporary work around.

**Comment by heloiselui** [comment]
Hey @nkn2022, I tested on my side and it’s working as expected. 

The issue was that `SideNavMenuItem` wasn’t rendering an interactive element, so it wasn’t tabbable. It becomes keyboard‑focusable when you provide an `href`.

Here is an updated example showing the correct usage:
https://stackblitz.com/edit/github-84wdmq-zcm8lham?file=src%2FApp.jsx&preset=node=

Docs: https://carbondesignsystem.com/components/UI-shell-left-panel/accessibility/
